### PR TITLE
ZBUG-3235: one char subdomain for the forwarding address is not allowed

### PR DESCRIPTION
**Pull request by ykomiyam**
[Bug description]
A forwarding address whose subdomain consists one character is rejected to input in the field of Admin Console -> Manage -> Accounts -> (pick one account) -> Forwarding -> User-spefieid forwarding addresses. "Invalid email address list" in red is displayed.

[Root cause]
The regex `AjxUtil.EMAIL_FULL_RE` for checking the ZimbraPrefMailForwardingAddress attribute had been set not to allow a single character subdomain.

According to the RFC 1034, the length of the subdomain is defined as "... is zero to 63 octets in length." (practically it's 1 to 63 octets), and most of the text fields of Admin Console accept such subdomain.

[Fix]
Changed the regex `AjxUtil.EMAIL_FULL_RE` to allow one character subdomain.

[Areas affected]
`AjxUtil.EMAIL_FULL_RE` affects not only to the check of `User- specified forwarding addresses` but also to the checks for various places in Admin Console as follows:

The following fields will accept a single character subdomain (no "Invalid email xxx" error will no longer be displayed) 
* Manage -> Accounts -> (pick an account) -> Forwarding -> User- specified forwarding addresses
* Manage -> Accounts -> New (Wizard) -> Forwarding -> User-specified forwarding addresses
* Manage -> Distribution Lists -> (pick one account) -> Or enter address below then click "Add" button

The following fields will be shown correctly (previously these are not shown on the screen at all):
* Manage -> Accounts -> (pick an account whose subdomain consists of a single character) -> Email at the top of the page (header area)
* Manage -> Resources -> (pick a resource whose subdomain consists of a single character) -> Email at the top of the page (header area) 

[Test]
* No "Invalid email xxx" error by a single character subdomain "i.example.com":
image
image
image

* Email fields which show an email with a single character subdomain "...@i.example.com" is displayed correctly:
image
image

### CODE REVIEW

**Comment by ykomiyam** [comment]
Additional test cases:
A. Test cases of one character subdomain:
Manage -> Accounts -> New (Wizard) -> Contact Information and put a manager address (an account with a single character subdomain `manager@i.example.com`) .  Verify that no error message is shown.  Then click "Finish", and verify that no  error message is shown.

B. Test cases of subdomains with two or longer characters
Manage -> Accounts -> (pick an account) -> Forwarding -> User- specified forwarding addresses and put an email whose subdomain consists of two characters, `user@ab.example.com`.  Verify that no error message is shown.

C. Test cases of subdomains with zero character (invalid subdomain name)
1. Manage -> Accounts -> (pick an account) -> Forwarding -> User- specified forwarding addresses and put an invalid email address format, `user@.example.com`.  Verify that error message is shown. 
2. Manage -> Accounts -> (pick an account) -> Contact Information and put an invalid email address format, `manager@.example.com` and save it.  Verify that error message is shown. 

image

image

image

image

File: WebRoot/js/ajax/util/AjxUtil.js [hunk 1]
```diff
@@ -69,7 +69,7 @@ AjxUtil.DOMAIN_NAME_FULL_RE = /^[A-Za-z0-9\-]{1,}(\.[A-Za-z0-9\-]{2,}){1,}$/;
 AjxUtil.HOST_NAME_RE = /^[A-Za-z0-9\-]{2,}(\.[A-Za-z0-9\-]{1,})*(\.[A-Za-z0-9\-]{2,})*$/;
 AjxUtil.HOST_NAME_WITH_PORT_RE = /^[A-Za-z0-9\-]{2,}(\.[A-Za-z0-9\-]{2,})*:([0-9])+$/;
 AjxUtil.EMAIL_SHORT_RE = /^[^@\s]+$/;
-AjxUtil.EMAIL_FULL_RE = /^[^@\s]+@[A-Za-z0-9\-]{2,}(\.[A-Za-z0-9\-]{2,})+$/;
+AjxUtil.EMAIL_FULL_RE = /^[^@\s]+@[A-Za-z0-9\-]{1,}(\.[A-Za-z0-9\-]{1,})+$/;
```
Comment (k-kato) [inline WebRoot/js/ajax/util/AjxUtil.js:72]: Do we need to change the length of top domain from `2` to `1`?
Is it reasonable to keep it as it is?
```suggestion
AjxUtil.EMAIL_FULL_RE = /^[^@\s]+@[A-Za-z0-9\-]{1,}(\.[A-Za-z0-9\-]{2,})+$/;
```

Comment (k-kato) [reply WebRoot/js/ajax/util/AjxUtil.js:72]: The above was wrong. Your code change looks good. It will support `a.b.domain.com`.

Comment (ykomiyam) [reply WebRoot/js/ajax/util/AjxUtil.js:72]: As Kato-san has mentioned above, this regex `/^[^@\s]+@[A-Za-z0-9\-]{1,}(\.[A-Za-z0-9\-]{1,})+$/` allows consequent one-letter subdomain names as well. 　
Example 1: `@a.b.domain.com`
image
Example 2: `@123.a.b.com`
image
The following examples are slightly unrealistic, but RFC-wise we are allowed such sub/top-level/domain names (see RFC 1034 Section 3.5, RFC-1123 section 2.1 as well):
Example 3: `@abc.1.2.3`
image

**Review by k-kato — APPROVED** [review]

**Comment by k-kato** [comment]
updated a commit comment

**Review by JayaG — APPROVED** [review]

**Review by silentsakky — APPROVED** [review]

**Pull request merged**